FWCore/Framework/interface/global/analyzerAbilityToImplementor.h

```diff
@@ -37,6 +37,11 @@ namespace edm {
         typedef edm::global::impl::StreamCacheHolder<edm::global::EDAnalyzerBase, C> Type;
       };
 
+      template <typename C>
+      struct AbilityToImplementor<edm::ProcessBlockCache<C>> {
```

Comment:
So the only difference between a ProcessBlockCache and a GlobalCache is another ProcessBlock can be activated if we close an output file during running?

Reply:
There will be a ProcessBlockCache for each ProcessBlock in the input that has any products in it. Probably we would clear them all when output files are closed, but even in the normal case with only one output file per output module there could be many ProcessBlockCache objects. The idea is input ProcessBlocks cannot be saved in memory throughout the process and only one input ProcessBlock is allowed in memory at a time. The ProcessBlockCache is a way to communicate some part of the ProcessBlock data to Events, Lumis, and Runs with the Framework keeping track of which ProcessBlocks (could be more than one) correspond to a particular Event, Lumi, or Run.